src/features/delegation/helpers/retrieve.ts

```diff
@@ -1,24 +1,21 @@
 import { DelegationAction } from '$/features/delegation'
 import networkPreflight from '$/utils/networkPreflight'
 import { providers, utils, Wallet } from 'ethers'
-import { call, put } from 'redux-saga/effects'
+import { call, cancelled, put } from 'redux-saga/effects'
 import jschkdf from 'js-crypto-hkdf'
 import isAuthorizedDelegatedAccount from '$/utils/isAuthorizedDelegatedAccount'
 import authorizeDelegatedAccount from '$/utils/authorizeDelegatedAccount'
-import { Controller } from '$/features/toaster/helpers/toast'
 import { ToastType } from '$/components/Toast'
-import retoast from '$/features/toaster/helpers/retoast'
 import recover from '$/utils/recover'
 import i18n from '$/utils/i18n'
 import getWalletProvider from '$/utils/getWalletProvider'
+import retoast from '$/features/toaster/helpers/retoast'
 
 export default function retrieve({
     owner,
 }: Pick<ReturnType<typeof DelegationAction.requestPrivateKey>['payload'], 'owner'>) {
     return call(function* () {
-        let tc: Controller | undefined
-
-        let dismissToast = false
+        const toast = retoast()
```

Comment:
No need to keep track on the `dismissToast` variable. `toast.open` is in charge of that now.